app/controllers/admin/api/registry/policies_controller.rb

```diff
@@ -32,13 +35,66 @@ def create
     respond_with current_account.policies.create(policy_params)
   end
 
+  ##~ e = sapi.apis.add
+  ##~ e.path = "/admin/api/registry/policies/{id}.json"
+  ##~ e.responseClass = "policy"
+  #
+  ##~ op             = e.operations.add
+  ##~ op.httpMethod  = "PUT"
+  ##~ op.summary     = "APIcast Policy Registry Update"
+  ##~ op.description = "Updates an APIcast Policy"
+  ##~ op.group       = "apicast_policies"
+  #
+  ##~ op.parameters.add @parameter_access_token
+  ##~ op.parameters.add @parameter_policy_id
+  ##~ op.parameters.add :name => "name", :description => "New name of the policy", :required => false, :dataType => "string", :paramType => "query"
+  ##~ op.parameters.add :name => "version", :description => "New version of the policy", :required => false, :dataType => "string", :paramType => "query"
+  ##~ op.parameters.add :name => "schema", :description => "New JSON Schema of the policy", :required => false, :dataType => "string", :paramType => "query"
+  def update
+    policy.update_attributes(policy_params)
+    respond_with(policy)
+  end
+
+  ##~ e = sapi.apis.add
+  ##~ e.path = "/admin/api/registry/policies/{id}.json"
+  ##~ e.responseClass = "policy"
+  #
+  ##~ op             = e.operations.add
+  ##~ op.httpMethod  = "GET"
+  ##~ op.summary     = "APIcast Policy Registry Read"
+  ##~ op.description = "Returns the APIcast policy by ID"
+  ##~ op.group       = "apicast_policies"
+  #
+  ##~ op.parameters.add @parameter_access_token
+  ##~ op.parameters.add @parameter_policy_id
+  def show
+    respond_with(policy)
+  end
+
   private
 
+  attr_reader :policy
+
   def authorize_policies
     authorize! :manage, :policy_registry
   end
 
   def policy_params
     params.require(:policy).permit(:name, :version, :schema)
   end
+
+  def policy_id_param
+    id = params[:id].to_s
+    path = URI.parse(request.original_url).path
+
+    if id.include?('.') && path.end_with?('json')
```

Comment:
Well this controller responds only to JSON, I do not understand this check 
https://github.com/3scale/porta/blob/857aeff1b344039d18adb44494a71b3285e043ef/app/controllers/admin/api/registry/policies_controller.rb#L7

Reply:
This is because of ActiveDocs. When the request is built and performed with the 3scale API Docs page, ".json" is appended to the path, making the route constraint to match it as part of the `:id` parameter.

Enhancing the route constraint with a better regex is hard because we cannot use anchors. `/([\w.-]+?)(?:\.(json|xml))?$/` for example would make sure that only the actual `id` is sliced. 

I refactored the controller method to use the regex.